gnd/src/main/java/com/google/android/gnd/persistence/local/room/entity/FeatureEntity.java

```diff
@@ -99,9 +116,24 @@ public static FeatureEntity fromMutation(FeatureMutation mutation, AuditInfo cre
             .setCreated(authInfo)
             .setLastModified(authInfo);
     mutation.getNewLocation().map(Coordinates::fromPoint).ifPresent(entity::setLocation);
+    mutation.getNewPolygonVertices().map(FeatureEntity::listToString)
+        .ifPresent(entity::setPolygonVertices);
     return entity.build();
   }
 
+  public static String listToString(ImmutableList<Point> vertices) {
+    if (vertices == null) {
```

Comment:
`getNewPolygonVertices()` optional never returns a null, and `vertices` is a non-null arg (no `@Nullable` declaration), so this check can be removed.